composer.json

```
@@ -21,13 +21,14 @@
 		}
 	],
 	"require": {
-		"php": "~7.4 | ~8.0",
-		"nette/application": "^2.4"
+		"php": "~7.4 | 8.0 - 8.3",
+		"nette/application": "^3.0"
```

Tady je otázka zdali ten přístup skrze magický setter, nebyl podporovaný už v 2.4. Pokud ano, tak by tady klidně mohly být obě verze, protože knihovně to je jedno a jde nám pouze o kompatibilitu s vyšší verzí nette/application při jejím použití.

```suggestion
		"nette/application": "^2.4 | ^3.0"
```

---

Dobrý nápad. Magický setter podporovaný v nette samozřejmě je.
Musel jsem upravit test, aby procházel v obou verzích. `nette/application` od 3.0 rozšiřuje metodu `TemplateFactory::createTemplate` o další parametr. Aby byl jeden test tak je třeba ověřovat správnost přes callback v Mockery.